cmd/salt-updater/main.go

```diff
@@ -126,28 +137,23 @@ func runMain() error {
 	return saltrequester.RunUpdate()
 }
 
-func runDbus() error {
+func runDbus() (*saltrequester.SaltState, error) {
 	//Read in previous state
-	saltState := &saltrequester.SaltState{}
-	data, err := ioutil.ReadFile(saltUpdateFile)
+	saltState, err := saltrequester.ReadStateFile()
 	if err != nil {
-		log.Printf("error reading previous salt state: %v", err)
-	} else if err := json.Unmarshal(data, saltState); err != nil {
-		log.Printf("error loading previous salt state: %v", err)
+		return nil, err
 	}
-
 	salt := &saltUpdater{
 		state: saltState,
 	}
 	go salt.modemConnectedListener()
 	if err := startService(salt); err != nil {
-		return err
+		return saltState, err
 	}
-	runtime.Goexit()
-	return nil
+	return saltState, err
 }
 
-func (s *saltUpdater) runSaltCallSync(args []string, makeEvent bool) (*saltrequester.SaltState, error) {
+func (s *saltUpdater) runSaltCallSync(args []string, updateCall bool, updateTime time.Time) (*saltrequester.SaltState, error) {
```

Comment:
Don't quite follow the renaming of `makeEvent` to `updateCall`

Reply:
we were only ever making an event when it was an update call, and i wanted to update LastUpdate if it was an update command. So rather than a parameter seemed easier to rename this.